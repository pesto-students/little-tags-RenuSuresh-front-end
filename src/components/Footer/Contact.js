import React from "react";
import { CONTACT_DETAILS } from "../../constant/properties";
import { useTranslation } from "react-i18next";
import "./Footer.css";

function Contact() {
  const [t] = useTranslation("common");
  return (
    <>
      <div className="footer__contact">
        <h3>{t(`footer.contactdetails`)}</h3>
        <pre>{CONTACT_DETAILS}</pre>
      </div>
    </>
  );
}

export default Contact;
