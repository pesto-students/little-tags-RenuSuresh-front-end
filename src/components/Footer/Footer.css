.footer {
  display: flex;
  justify-content: space-around;
  padding-top: 1em;
  height: 20vh;
  background-color: var(--clr-primary);
  color: white;
  margin-top: auto;
  font-size: small;
}
.footer__contact > h3,
.footer__quicklinks > h3,
.footer__terms > h3 {
  margin-bottom: 2ch;
  text-transform: uppercase;
}

.footer__quicklinks > ul,
.footer__terms > ul {
  list-style: none;
  margin-left: -2.4rem;
}
.footer__terms > ul {
  text-transform: capitalize;
}
li {
  margin-bottom: 5px;
}
@media only screen and (max-width: 426px) {
  .footer {
    display: flex;
    flex-direction: column;
    height: fit-content;
    padding-left: 4rem;
    padding-bottom: 4rem;
  }
  .footer__contact,
  .footer__quicklinks,
  .footer__terms {
    margin-top: 2rem;
  }
}
