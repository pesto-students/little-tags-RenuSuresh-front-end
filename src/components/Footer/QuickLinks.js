import React from "react";
import { useTranslation } from "react-i18next";
import "./Footer.css";

function QuickLinks() {
  const [t] = useTranslation("common");

  return (
    <>
      <div className="footer__quicklinks">
        <h3>{t(`footer.quickLinks`)}</h3>
        <ul className="footer__contact__details">
          <li>{t("header.category.men")}</li>
          <li> {t("header.category.women")}</li>
          <li>{t("header.category.electronics")}</li>
        </ul>
      </div>
    </>
  );
}

export default QuickLinks;
