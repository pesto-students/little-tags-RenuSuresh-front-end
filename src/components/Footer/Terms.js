import React from "react";
import { useTranslation } from "react-i18next";
import "./Footer.css";

function Terms() {
  const [t] = useTranslation("common");

  return (
    <>
      <div className="footer__terms">
        <h3>{t(`footer.policy`)}</h3>

        <ul>
          <li>{t(`footer.terms`)}</li>
          <li>{t(`footer.shipping`)}</li>
          <li>{t(`footer.cancel`)}</li>
          <li>{t(`footer.privacy`)}</li>
        </ul>
      </div>
    </>
  );
}

export default Terms;
