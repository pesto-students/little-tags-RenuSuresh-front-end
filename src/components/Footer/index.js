import { Grid } from "@material-ui/core";
import React from "react";
import "./Footer.css";
import Contact from "./Contact";
import Terms from "./Terms";
import QuickLinks from "./QuickLinks";

function FooterIndex() {
  return (
    <div className="footer">
      <Contact />
      <Terms />
      <QuickLinks />
    </div>
  );
}

export default FooterIndex;
