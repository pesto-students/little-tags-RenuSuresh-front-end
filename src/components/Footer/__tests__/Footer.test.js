import { fireEvent, render, cleanup } from "@testing-library/react";
import React from "react";
import Enzyme from "enzyme";
import renderer from "react-test-renderer";
import Footer from "../index";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { I18nextProvider } from "react-i18next";
import i18next from "i18next";
import QuickLinks from "../QuickLinks";
import lang_en from "../../../locales/en/common.json";
import lang_hi from "../../../locales/hi/common.json";

Enzyme.configure({ adapter: new Adapter() });

const { mount } = Enzyme;

describe("<Footer />", () => {
  it("should render snapshot", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
