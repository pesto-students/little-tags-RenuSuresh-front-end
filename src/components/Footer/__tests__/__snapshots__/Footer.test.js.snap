// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render snapshot 1`] = `
<div
  className="footer"
>
  <div
    className="footer__contact"
  >
    <h3>
      footer.contactDetails
    </h3>
    <pre
      className="footer__contact__details"
    >
      9 , Shivaji Colony, Opp Guddy Bar,
Andheri Kurla Rd, Chakala,
Andheri (west),
Mumbai
Maharashtra- 400099
02228232309
    </pre>
  </div>
  <div
    className="footer__terms"
  >
    <h3>
      footer.policy
    </h3>
    <ul
      className="footer__contact__details"
    >
      <li>
        footer.terms
      </li>
      <li>
        footer.shipping
      </li>
      <li>
        footer.cancel
      </li>
      <li>
        footer.privacy
      </li>
    </ul>
  </div>
  <div
    className="footer__quicklinks"
  >
    <h3>
      footer.quickLinks
    </h3>
    <ul
      className="footer__contact__details"
    >
      <li>
        header.category.men
      </li>
      <li>
         
        header.category.women
      </li>
      <li>
        header.category.electronics
      </li>
    </ul>
  </div>
</div>
`;
